### PR TITLE
Add a Partner Cookbook badge to cookbooks

### LINKED ISSUE #1175: Chef Partner Cookbooks need an icon on the cookbook page
There should be a visual indicator for cookbooks that have gone through the Chef Partner Cookbook process.

In support of ROAD-73 (Display quality metrics for each cookbook on the Supermarket)
screenshot 2015-12-03 15 38 46

### LINKED ISSUE #1176: Chef Partner Cookbooks need an icon on the cookbook list
There should be a visual indicator for cookbooks that have gone through the Chef Partner Cookbook process when listing cookbooks through a search.

In support of ROAD-73 (Display quality metrics for each cookbook on the Supermarket)
screenshot 2015-12-03 15 43 28

### LINKED ISSUE #1178: There should be a rake task for adding Chef Partner Cookbooks
Because Chef Partner Cookbooks will be infrequently added, to support the MVP we'll look to add them via the CLI rather than build a UI for them. Please add a rake task for inserting them into the database properly.

In support of ROAD-73 (Display quality metrics for each cookbook on the Supermarket)

### PR DESCRIPTION
**Pull request by jjasghar**
fixes #1178, fixes #1175, fixes #1176 

This PR allows for the ability to change a cookbook with a rake task to mark it as a Partner Cookbook. We need to change the image to what it needs to be, not me giving the thumbs up. This will be WIP until that is done.

TODO:
- [X] include the changes to `db/schema.rb` resulting from the new migration
- [X] add spec(s) for `Badgeable` concern and that `Cookbook` should behave like something badgeable
- [x] add a service/process class (and tests for it) for adding a badge to a cookbook to move logic out of the rake task
- [x] add feature spec(s) around the user experience of seeing a partner badge on a partner cookbook (implies there is a view for badges on a cookbook)

### CODE REVIEW

**Comment by mattray** [comment]
Woohoo!
:dancers: :evergreen_tree: :meat_on_bone:

**Comment by jjasghar** [comment]
@robbkidd the failed check here, i'm not sure how to fix it?

**Comment by robbkidd** [comment]
A-ha. That's where I didn't tell you about checking in the change to `db/schema.rb` along with the migration script added to `db/migrate`. [Travis is told to run `db:schema:load`](https://github.com/chef/supermarket/blob/master/.travis.yml#L12) which depends on that `schema.rb` being complete and does _not_ use the actual migration script.

**Comment by robbkidd** [comment]
I have an idea I would like to spike on around adding this feature as a generic cookbook badging feature of which "partner" is the first type.

**Comment by jjasghar** [comment]
image

**Comment by robbkidd** [comment]
Current state:

### Partner Badge in cookbook sidebar
badge in cookbook sidebar

### Partner Badge in cookbook list
badge in cookbook list

**Comment by robbkidd** [comment]
Note: screenshots look good (to me), but we need tests to make this shippable. Working on them next.

**Comment by nathenharvey** [comment]
I'm :+1: on the visual treatment of partner cookbooks.

I do want to understand where the icon will link to though.

**Comment by robbkidd** [comment]
Lordy. OK, all, I think this is ready.

@nellshamrell @jjasghar :eyes:

The [screenshots above](https://github.com/chef/supermarket/pull/1199#issuecomment-173349747) are still accurate.

**Comment by jjasghar** [comment]
image

File: app/models/cookbook.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class Cookbook < ActiveRecord::Base
+  include Badgeable
```
Comment (nellshamrell) [inline app/models/cookbook.rb]: Is this module used anywhere in the cookbook model itself?

Comment (robbkidd) [inline app/models/cookbook.rb]: Nope.

Comment (nellshamrell) [inline app/models/cookbook.rb]: Should we take this line out?

Comment (robbkidd) [inline app/models/cookbook.rb]: I don't think so. Including the concern is what gives an instance of cookbook the methods called on it in the view for showing badges (e.g. `.badges`, `.is? :partner`).

Comment (nellshamrell) [inline app/models/cookbook.rb]: Cool

**Comment by nellshamrell** [comment]
:shipit:

**Pull request merged**